src/Razor/benchmarks/Microsoft.AspNetCore.Razor.Microbenchmarks/LanguageServer/RazorSemanticTokensRangeEndpointBenchmark.cs

```diff
@@ -0,0 +1,203 @@
+
+public class RazorSemanticTokensRangeEndpointBenchmark : RazorLanguageServerBenchmarkBase
+{
+    private IRazorSemanticTokensInfoService RazorSemanticTokenService { get; set; }
+
+    private SemanticTokensRangeEndpoint SemanticTokensRangeEndpoint { get; set; }
+
+    private DocumentVersionCache VersionCache { get; set; }
+
+    private Uri DocumentUri => DocumentContext.Uri;
+
+    private IDocumentSnapshot DocumentSnapshot => DocumentContext.Snapshot;
+
+    private VersionedDocumentContext DocumentContext { get; set; }
+
+    private Range Range { get; set; }
+
+    private new IRazorLogger Logger { get; set; }
+
+    private ProjectSnapshotManagerDispatcher ProjectSnapshotManagerDispatcher { get; set; }
+
+    private string PagesDirectory { get; set; }
+
+    private string ProjectFilePath { get; set; }
+
+    private string TargetPath { get; set; }
+
+    private static int NumberOfCsSemanticRangesToReturn { get; set; }
+
+    [GlobalSetup]
+    public async Task InitializeRazorSemanticAsync()
+    {
+        EnsureServicesInitialized();
+
+        var projectRoot = Path.Combine(RepoRoot, "src", "Razor", "test", "testapps", "ComponentApp");
+        ProjectFilePath = Path.Combine(projectRoot, "ComponentApp.csproj");
+        PagesDirectory = Path.Combine(projectRoot, "Components", "Pages");
+        var filePath = Path.Combine(PagesDirectory, $"SemanticTokens.razor");
+        TargetPath = "/Components/Pages/SemanticTokens.razor";
+
+        var documentUri = new Uri(filePath);
+        var documentSnapshot = GetDocumentSnapshot(ProjectFilePath, filePath, TargetPath);
+        var version = 1;
+        DocumentContext = new VersionedDocumentContext(documentUri, documentSnapshot, version);
+        Logger = new NoopLogger();
+        SemanticTokensRangeEndpoint = new SemanticTokensRangeEndpoint();
+
+        var text = await DocumentContext.GetSourceTextAsync(CancellationToken.None).ConfigureAwait(false);
+        Range = new Range
+        {
+            Start = new Position { Line = 0, Character = 0 },
+            End = new Position { Line = text.Lines.Count - 1, Character = text.Lines.Last().Span.Length - 1 }
+        };
+    }
+
+    [Benchmark(Description = "Razor Semantic Tokens Range Endpoint [0]")]
+    public async Task RazorSemanticTokensRangeEndpointZeroCsRangesAsync()
+    {
+        var textDocumentIdentifier = new TextDocumentIdentifier { Uri = DocumentUri };
+        var cancellationToken = CancellationToken.None;
+        var documentVersion = 1;
+
+        await UpdateDocumentAsync(documentVersion, DocumentSnapshot, cancellationToken).ConfigureAwait(false);
+
+        var request = new SemanticTokensRangeParams { Range = Range, TextDocument = textDocumentIdentifier };
+
+        var languageServer = RazorLanguageServer.GetInnerLanguageServerForTesting();
+        var requestContext = new RazorRequestContext(DocumentContext, Logger, languageServer.GetLspServices());
+
+        NumberOfCsSemanticRangesToReturn = 0;
```

Comment:
If you mark this property with `[Params(0, 100, 1000)]` you can remove the duplicated benchmark methods, and just have the one. BenchmarkDotNet will run it three times, with the right values.

You'll have to make it non-static, and pass it through to the your service though

Reply:
Oh! I had no idea. Will do